src/Components/Shifting/CommentsSection.tsx

```diff
@@ -73,40 +58,67 @@ const CommentSection = (props: CommentSectionProps) => {
         </ButtonV2>
       </div>
       <div className=" w-full">
-        {isLoading ? (
+        {loading ? (
           <CircularProgress />
         ) : (
-          comments.map((comment: any) => (
-            <div
-              key={comment.id}
-              className="mt-4 flex w-full flex-col rounded-lg border border-gray-300 bg-white p-4 text-gray-800"
-            >
-              <div className="flex  w-full ">
-                <p className="text-justify">{comment.comment}</p>
-              </div>
-              <div className="mt-3">
-                <span className="text-xs text-gray-500">
-                  {comment.modified_date
-                    ? formatDateTime(comment.modified_date)
-                    : "-"}
-                </span>
-              </div>
-              <div className=" mr-auto flex items-center rounded-md border bg-gray-100 py-1 pl-2 pr-3">
-                <div className="flex h-8 w-8 items-center justify-center rounded-full bg-primary-700 p-1 uppercase text-white">
-                  {comment.created_by_object?.first_name?.charAt(0) ||
-                    t("unknown")}
+          <PaginatedList
+            route={routes.getShiftComments}
```

Comment:
You are loading shift comments twice, one just for the loading state, and other just for the loading + comments. Why?

Reply:
When I add a new comment the PaginatedList is not refreshing i.e refetch if but I manually refetch in another usequey it's working

Reply:
When a new comment is added, the paginated list does not handle it, meaning that the state remains the same, if not can I use useState to create a boolean variable that toggles between true and false. Initially, it should be set to false, and when a new comment is added, can set it to true. After refetching the data to update the paginated list, set it back to false.

Reply:
No, the correct way is to call the `refetch` of the `useQuery` managed by the `PaginatedList` itself somehow. Feel free to tinker the PaginatedList or the context to achieve this.

Reply:
@rithviknishad 
I have added the refetch logic inside the PaginatedList itself.could you please check it

Reply:
Nice work!